Allow passing an optional ARN when health checking  an AWSOIDC integration

kimlisa opened the pull request:
This new addition allows us to health check aws oidc integrations before a user creates or edits the integration.


Demo:

`curl --data '{"arn":"arn:aws:iam::123456789012:role/MyRole"}' '.../integrations/aws-oidc/my-integration/ping'`
```json
{
  "accountId": "123456789012",
  "arn": "arn:aws:sts::123456789012:assumed-role/MyRole/123",
  "userId": "AAA:17290"
}
```

GavinFrazar commented:
Is the check to see if the role ARN exists (or if it's configured to trust the OIDC token) before creating the integration?

marcoandredinis commented:
I'll convert it into draft because I want to polish some parts and add tests.

marcoandredinis commented:
> Is the check to see if the role ARN exists (or if it's configured to trust the OIDC token) before creating the integration?

It checks if it is configured to trust the OIDC token.
So, it will fail when the role does not exist or when the IAM Role's trust relationship does not allow this particular Identity Provider's.

marcoandredinis committed:
AWS OIDC Ping: allow custom ARN

This PR changes the Ping method to accept a custom ARN.

This is meant to be used by WebUI to do a health check for the
integration:
- when creating
- when editing
- when selecting during Discover flows

If the Ping method receives an ARN, it will use that value instead of
using the one stored in the backend.

marcoandredinis committed:
rename arn to roleArn

marcoandredinis commented:
@fspmarshall @camscale Can you please take a look?

camscale reviewed (approved):
Sorry I didn't review this sooner. When I see a PR with people explicitly chosen to review it, I kind of ignore the bot assignment assuming those others have better context for the review.

marcoandredinis commented on lib/integrations/awsoidc/clients_test.go:
```diff
@@ -29,30 +29,30 @@ import (
-			Token:           "token",
-			RoleARN:         "some-arn",
-			Region:          "?",
+			//IntegrationName: "my-integration",
```
Removed, thank you.

marcoandredinis committed:
rename arn to role_arn

marcoandredinis commented:
@kimlisa After changing some things, would you mind taking a look?
Given that you created the PR, you can't actually approve it but let me know if it looks good and I'll ask for a G1 bot review.

kimlisa commented:
approved

r0mant reviewed (approved):
Bot.
